src/deviceTree.ts

```diff
@@ -29,84 +32,46 @@ export class DeviceTree implements vscode.TreeDataProvider<vscode.TreeItem> {
         }
     }
 
-    public getTreeItem(element: vscode.TreeItem): vscode.TreeItem {
-        return element;
+    public getTreeItem(element: INode): Promise<vscode.TreeItem> | vscode.TreeItem  {
+        return element.getTreeItem();
     }
 
-    public async getChildren(element?: vscode.TreeItem): Promise<vscode.TreeItem[]> {
+    public async getChildren(element?: INode): Promise<INode[]> {
         let iotHubConnectionString = await Utility.getConnectionString(Constants.IotHubConnectionStringKey, Constants.IotHubConnectionStringTitle, false);
         if (!iotHubConnectionString) {
-            return this.getDefaultTreeItems();
+            return Utility.getDefaultTreeItems();
         }

```

Comment:
Put this code block into `if (!element)` code block?

Reply:
In case use remove connection string when load, so we put it  outside the element